core/templates/dev/head/services/ExplorationHtmlFormatterService.js

```diff
@@ -52,6 +52,12 @@ oppia.factory('ExplorationHtmlFormatterService', [
         if (labelForFocusTarget) {
           element.attr('label-for-focus-target', labelForFocusTarget);
         }
+        // We cannot restrict this attr to interactions with
+        // show_nav_submit_button because of special cases such
+        // as ItemSelectionInput.
```

Comment:
Also, let's specify this new attr only for interactions that have show_submit_button_in_nav, plus ItemSelectionInput. We should also put a comment in both places saying that these two things should match up with each other.

In addition, I think we should specify the setAnswerValidity function in the isolate scope for all interactions in which that functionality is provided. It should become part of the standard infrastructure for those interactions. To validate that what you've done generalizes beyond just ItemSelection, you might want to implement disabling of the submit button for either number/text -- this should be straightforward to do, and it provides assurance that what you have here works in general. We may also want to add a comment above that scope attr to say something like "Devs should ensure that this is called every time the answer is changed, even prior to submission." -- since otherwise I can see them forgetting to do it.

Reply:
Sure! For Something like TextInput, what would the criteria for valid be, non-empty? Isn't it technically possible that the answer is an empty string?

Reply:
Yeah, let's go with non-empty. In all the years I've never seen an empty string being a valid answer.

Reply:
Got it.